apps/functions-subscription/src/adapters/azure/functions/list-trials.ts

```diff
@@ -0,0 +1,70 @@
+import * as H from '@pagopa/handler-kit';
+import { pipe, flow } from 'fp-ts/lib/function';
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import * as E from 'fp-ts/Either';
+import * as O from 'fp-ts/lib/Option';
+import * as t from 'io-ts';
+import { httpAzureFunction } from '@pagopa/handler-kit-azure-func';
+import { TrialPaginatedCollection } from '../../../generated/definitions/internal/TrialPaginatedCollection';
+import { SystemEnv } from '../../../system-env';
+import { getAndValidateUser, parseQueryParameter } from './middleware';
+import { toHttpProblemJson } from './errors';
+import { toTrialListAPI } from './codec';
+import { TrialIdCodec } from '../../../domain/trial';
+import {
+  IntegerFromString,
+  WithinRangeInteger,
+} from '@pagopa/ts-commons/lib/numbers';
+
+type Env = Pick<SystemEnv, 'listTrials'>;
+
+export const QueryPageSize = t.union([
+  t.undefined,
+  IntegerFromString.pipe(WithinRangeInteger(1, 100)),
+]);
```

Comment:
**suggestion**: add a short comment to explain those numbers

Reply:
Added comments in 135e4c218e4d43aeae3ea6cb5dff5f2bf85a5fe9